drivers/adc/adc_stm32.c

```diff
@@ -802,13 +802,21 @@ static void adc_stm32_setup_channels(const struct device *dev, uint8_t channel_i
 {
 	const struct adc_stm32_cfg *config = dev->config;
 
+#ifdef CONFIG_SOC_SERIES_STM32G4X
 	if (config->has_temp_channel &&
-		__LL_ADC_CHANNEL_TO_DECIMAL_NB(ADC_CHANNEL_TEMPSENSOR) == channel_id) {
+	    ((__LL_ADC_CHANNEL_TO_DECIMAL_NB(LL_ADC_CHANNEL_TEMPSENSOR_ADC1) == channel_id) ||
```

Comment:
Don't you also have to check the ADC index as well, when the defines are different depending on the ADC? Otherwise I suppose an issue could be somebody calling this function e.g. for ADC1 with LL_ADC_CHANNEL_TEMPSENSOR_ADC5?

Reply:
Right, will fix it